### ISSUE TITLE
Switch from tag-based release flow to push-based release flow

### ISSUE THREAD

**Issue by mcevoypeter** [urbit/vere#47]
To simplify the release flow, we should switch from a tag-based flow to a push-based release flow. In the new scheme, we'll have a `release` branch, which has all work that is tested, reviewed, and ready to go out in the next release. All feature work will target the `release` branch. Any time a commit is pushed to a feature branch--any branch that is not `master` or `release`--all unit and integration tests will run via the [feature flow](./.github/workflows/feature.yml). Once a feature is ready to be staged for release, the feature's PR will merge into `release`, at which point the [pre-release flow](./.github/workflows/pre-release.yml) will run, triggering binary uploads for the `soon` train to bootstrap.urbit.org. Once a release is ready to go out, the version number in `pkg/vere/VERSION` will be bumped and `release` will be rebased on top of `master`, triggering the [release flow](./.github/workflows/release.yml) to run, which will re-run all unit and integration tests and--assuming the tests pass--trigger binary uploads for the `live` train to bootstrap.urbit.org.
